packages/server/src/services/compose.ts

```diff
@@ -437,13 +437,26 @@ export const rebuildRemoteCompose = async ({
 	return true;
 };
 
-export const removeCompose = async (compose: Compose) => {
+export const removeCompose = async (
+	compose: Compose,
+	deleteVolumes: boolean,
+) => {
 	try {
 		const { COMPOSE_PATH } = paths(!!compose.serverId);
 		const projectPath = join(COMPOSE_PATH, compose.appName);
 
+		console.log("API: DELETE VOLUMES=", deleteVolumes);
+
 		if (compose.composeType === "stack") {
-			const command = `cd ${projectPath} && docker stack rm ${compose.appName} && rm -rf ${projectPath}`;
+			const listVolumesCommand = `docker volume ls --format \"{{.Name}}\" | grep ${compose.appName}`;
+			const removeVolumesCommand = `${listVolumesCommand} | xargs -r docker volume rm`;
+			let command: string;
+			if (deleteVolumes) {
+				command = `cd ${projectPath} && docker stack rm ${compose.appName} && ${removeVolumesCommand} && rm -rf ${projectPath}`;
+			} else {
+				command = `cd ${projectPath} && docker stack rm ${compose.appName} && rm -rf ${projectPath}`;
+			}
+
```

Comment:
I think it is not necessary for docker stack, this would be almost the equivalent of the applications only in compose format, I think let's just apply it to compose.

Reply:
I'll have it done by today :)